ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/AbstractDebugTraceBlock.java

```diff
@@ -0,0 +1,157 @@
+              .orElse(TraceOptions.DEFAULT);
+    } catch (JsonRpcParameter.JsonRpcParameterException e) {
+      throw new InvalidJsonRpcParameters(
+          "Invalid transaction trace parameter (index 1)",
+          RpcErrorType.INVALID_TRANSACTION_TRACE_PARAMS,
+          e);
+    }
+    return traceOptions;
+  }
+
+  protected Collection<DebugTraceTransactionResult> getTraces(
+      final JsonRpcRequestContext requestContext,
+      final TraceOptions traceOptions,
+      final Optional<Block> maybeBlock) {
+    return maybeBlock
+        .flatMap(
+            block ->
+                Tracer.processTracing(
+                    getBlockchainQueries(),
+                    block.getHash(),
+                    traceableState -> {
+                      Collection<DebugTraceTransactionResult> tracesList =
+                          new CopyOnWriteArrayList<>();
+                      final ProtocolSpec protocolSpec =
+                          protocolSchedule.getByBlockHeader(block.getHeader());
+                      final MainnetTransactionProcessor transactionProcessor =
+                          protocolSpec.getTransactionProcessor();
+                      final TraceBlock.ChainUpdater chainUpdater =
+                          new TraceBlock.ChainUpdater(traceableState);
+
+                      TransactionSource transactionSource = new TransactionSource(block);
+                      DebugOperationTracer debugOperationTracer =
+                          new DebugOperationTracer(traceOptions, true);
+                      ExecuteTransactionStep executeTransactionStep =
+                          new ExecuteTransactionStep(
+                              chainUpdater,
+                              transactionProcessor,
+                              getBlockchainQueries().getBlockchain(),
+                              debugOperationTracer,
+                              protocolSpec,
+                              block);
+                      DebugTraceTransactionStep debugTraceTransactionStep =
+                          new DebugTraceTransactionStep();
+                      Pipeline<TransactionTrace> traceBlockPipeline =
+                          createPipelineFrom(
+                                  "getTransactions",
+                                  transactionSource,
+                                  4,
+                                  outputCounter,
+                                  false,
+                                  "debug_trace_block")
+                              .thenProcess("executeTransaction", executeTransactionStep)
+                              .thenProcessAsyncOrdered(
+                                  "debugTraceTransactionStep", debugTraceTransactionStep, 4)
+                              .andFinishWith("collect_results", tracesList::add);
+
+                      try {
+                        if (getBlockchainQueries().getEthScheduler().isPresent()) {
```

Comment:
Is there a realistic scenario where EthScheduler isn't available?

Reply:
This was inherited from Trace_block implementation. I changed both Trace* and DebugTrace* implementations to use the Besu ethScheduler, created when we start Besu. Now, we don't need anymore to check if it null or not.